### PR TITLE
run openapi-linter on all files

### PR DESCRIPTION
**Pull request by ghost**
Run `openapi-linter`/`asyncapi-linter` when source-of-truth OpenAPI/AsyncAPI document changes.

### CODE REVIEW

**Review by ls-jad-elkik — APPROVED** [review]

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -312,10 +312,17 @@
...
 - id: openapi-linter
-  name: Lint OpenAPI documents
-  description: Lint OpenAPI documents with spectral
-  entry: 809245501444.dkr.ecr.eu-central-1.amazonaws.com/openapi-linter:release-latest-main
+  name: lint openapi documents
+  description: lints openapi documents with spectral
+  entry: >
+    809245501444.dkr.ecr.us-east-1.amazonaws.com/release/internal/image/openapi-linter:21-73a3f80
+    openapi
   language: docker_image
```
Comment (ls-samer-abuthaher) [inline .pre-commit-hooks.yaml:328]: I think we should add the files so it's only sensitive to openapi no?

**Review by ls-jad-elkik — DISMISSED** [review]
We should have the `files: regex-here` and `pass_filenames: true` too

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -312,10 +312,19 @@
...
     entry: ls_pre_commit_hooks/plantuml-renderer.sh
     files: '.*\.puml$'
+- id: asyncapi-linter
+  name: lint asyncapi documents
+  description: lints asyncapi documents with spectral
+  entry: >
+    809245501444.dkr.ecr.us-east-1.amazonaws.com/release/internal/image/openapi-linter:21-73a3f80
+    asyncapi
+  language: docker_image
+  files: ^apidoc/\.+/asyncapi\.yaml$
```
Comment (ls-samer-abuthaher) [inline .pre-commit-hooks.yaml:322]:
```suggestion
  files: ^apidoc/.+/asyncapi\\.yaml$
```

File: .pre-commit-hooks.yaml [hunk 3]
```diff
@@ -312,10 +312,19 @@
...
-  entry: 809245501444.dkr.ecr.eu-central-1.amazonaws.com/openapi-linter:release-latest-main
+  name: lint openapi documents
+  description: lints openapi documents with spectral
+  entry: >
+    809245501444.dkr.ecr.us-east-1.amazonaws.com/release/internal/image/openapi-linter:21-73a3f80
+    openapi
   language: docker_image
-  pass_filenames: false
-  files: '^apidoc/[^/]+/(openapi|asyncapi)\\.yaml$'
+  files: ^apidoc/\.+/openapi\.yaml$
```
Comment (ls-samer-abuthaher) [inline .pre-commit-hooks.yaml:330]:
```suggestion
  files: ^apidoc/.+/openapi\\.yaml$
```

**Review by ls-adrian-chong — APPROVED** [review]

**Pull request merged**